spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteManifestsSparkAction.java

```diff
@@ -264,14 +293,15 @@ private <T, U> U withReusableDS(Dataset<T> ds, Function<Dataset<T>, U> func) {
     }
   }
 
-  private List<ManifestFile> findMatchingManifests() {
+  private List<ManifestFile> findMatchingManifests(ManifestContent content) {
     Snapshot currentSnapshot = table.currentSnapshot();
 
     if (currentSnapshot == null) {
       return ImmutableList.of();
     }
 
-    return currentSnapshot.dataManifests(table.io()).stream()
+    return currentSnapshot.allManifests(table.io()).stream()
+        .filter(manifest -> manifest.content() == content)
```

Comment:
nit: we can combine the filter conditions with below one?

Reply:
I kept it separate to avoid Spotless formatting the closure in a weird way. I can define a method and call it here. These filters are chained together prior to execution.

Reply:
I restructured it a bit, no longer applies.